File: FlowerShop/models/dto/Flower.cs
using System.ComponentModel.DataAnnotations;

namespace FlowerShop.models.dto
{
    public class Flower
    {
        //public Category category;
        [Key]
        public int id { get; set; }
        public string name { get; set; }
        public string description { get; set; }
        public string shortDescription { get; set; }
        //public Price priceDto { get; set; }
        public string photo { get; set; }
        public bool thumbnail { get; set; }
        public bool inCart { get; set; }
    }
}

File: FlowerShop/client/src/app/classes/flower.ts
import { SafeUrl } from '@angular/platform-browser';

export interface PriceDto {
  id: number;
  price: number;
  date: string;
  itemId: number;
}

export interface Category {
  id: number;
  name: string;
  description: string;
  thumbnail: string;
  photo: string;
}

export interface Flower {
  category: Category;
  id: number;
  name: string;
  description: string;
  shortDescription: string;
  priceDto: PriceDto;
  photo: string;
  loadedPhoto: SafeUrl;
  thumbnail: boolean;
  inCart: boolean;
}

export interface FlowersResponseDto {
  response: any;
  flowers: Flower[];
}
